engine/src/main/java/org/apache/apex/engine/YarnAppLauncherImpl.java

```diff
@@ -59,6 +61,7 @@
   @Override
   public YarnAppHandleImpl launchApp(final StreamingApplication app, Configuration conf, Attribute.AttributeMap launchParameters) throws LauncherException
   {
+    this.conf = conf;
```

Comment:
Please see my other comment, likely it is better to cache yarnClient itself instead of conf.